lib/valid_email2/dns.rb

```diff
@@ -26,6 +26,26 @@ def mx_servers(domain)
       fetch(domain, Resolv::DNS::Resource::IN::MX)
     end
 
+    def mx_servers_disposable?(domain, domain_list)
+      servers = mx_servers(domain)
+      cache_key = generate_mx_cache_key(domain, domain_list, servers)
+      return MX_SERVERS_CACHE[cache_key] if !cache_key.nil? && MX_SERVERS_CACHE.key?(cache_key)
+
+      result = servers.any? do |mx_server|
+        return false unless mx_server.respond_to?(:exchange)
+
+        mx_server = mx_server.exchange.to_s
+
+        domain_list.any? do |disposable_domain|
+          mx_server.end_with?(disposable_domain) && mx_server.match?(/\A(?:.+\.)*?#{disposable_domain}\z/)
```

Comment:
o_O If you need to check that `mx_server` is sub-domain of `disposable_domain`, there's a bit cheaper version:

```suggestion
          mx_server.end_with?(".#{disposable_domain}") && mx_server.size > disposable_domain.size + 1
```